src/core/Track.cpp

```diff
@@ -1070,12 +1112,56 @@ void TrackContentObjectView::mouseReleaseEvent( QMouseEvent * me )
 	{
 		setSelected( !isSelected() );
 	}
-
-	if( m_action == Move || m_action == Resize || m_action == ResizeLeft )
+	else if( m_action == Move || m_action == Resize || m_action == ResizeLeft )
 	{
 		// TODO: Fix m_tco->setJournalling() consistency
 		m_tco->setJournalling( true );
 	}
+	else if( m_action == Split )
+	{
+		SampleTCO * leftTCO = dynamic_cast<SampleTCO*>( m_tco );
+
+		leftTCO->setMarkerEnabled( false );
+
+		const int relativePixelPos = me->pos().x();
+		const float ppt = m_trackView->trackContainerView()->pixelsPerTact();
+		MidiTime splitPos = relativePixelPos * MidiTime::ticksPerTact() / ppt;
+		const float snapSize = gui->songEditor()->m_editor->getSnapSize();
+
+		//Essentially duplicate logic from knifeMarkerPos, but shift needs to be inverted for some reason
+		if ( me->modifiers() & Qt::ControlModifier
+		  || me->modifiers() & Qt::AltModifier    ){}
```

Comment:
This looks a little bit odd to me. Perhaps you find a better way for this logic?